COMPOUNDING_RATES.md

```diff
@@ -0,0 +1,78 @@
+# Terms 
+
+## exchangeRate
+
+A compounding token's exchange rate determines how much underlying the compounding token is currently worth, or in other terms, how many base units of underlying can 1 compounding token be exchanged for.
+
+## supplyRate - apy/interestRate
+
+A compounding token's supply rate is essentially the token's (compounded) lending interest rate. Depending on the protocol, this rate can be expressed in terms of rate per block (i.e. compound, rari), per second (i.e. Aave, Euler), or per year (Yearn?). In general this rate should be annualized to make it comparable to traditional APYs.
+
+# Protocols
+
+## Compound & Rari
+
+https://compound.finance/docs/ctokens#exchange-rate
+
+https://compound.finance/docs#protocol-math
+
+```ts
+// normalize the exchange rate to the decimals of the underlying
+const exchangeRate = cToken.exchangeRate && fixed(contractAmount(cToken.exchangeRate, 18 - cToken.decimals + underlying.decimals)).toString(),
```

Comment:
is it 18 because the unit is wei or some inferenced number of decimals?

former seems ok, latter would not?

Reply:
good catch^^ so this one comes straight from compounds exchangeRate docs (https://compound.finance/docs/ctokens#exchange-rate). it describes the return value as follows:

> `RETURN`: The current exchange rate as an unsigned integer, scaled by 1 * 10^(18 - 8 + Underlying Token Decimals).

that's where the 18 comes from.

it's important to note though, that this section in the markdown file used the calculations from the frontend repo, and is not actually implemented in swivel-js. due to how `libCompounding` in our contracts doesn't perform any scaling of the exchange rate (except for euler and erc-4626 where we pass 1e+26 to the convertToAssets methods....) swivel-js won't either. meaning each exchange rate will have a scale depending on the protocol. that's not optimal, but i can't change it, as swivel is gonna store the unscaled exchange rates in the vault. so swivel-js returns the same scale as the one used in the contracts.